src/AutoRest.CSharp/Common/Generation/Writers/SerializationWriter.cs

```diff
@@ -201,7 +201,7 @@ private void WriteJsonDeserialize(CodeWriter writer, SchemaObjectType model, Jso
                     }
                 }
 
-                if (model.Declaration.IsAbstract)
+                if (model.Declaration.IsAbstract || model.IsAbstract)
```

Comment:
Now the property `IsAbstract` is defined in `SchemaObjectType`. But I believe it makes sense that we move it to `TypeProvider` since if a `TypeProvider` provides a type, it makes perfect sense that it is providing an abstract type (class).
And we could change the function `BuilderHelpers.CreateTypeAttributes` to consume a new parameter `isAbstract` so that these two calls will align with each other

Reply:
Agreed. If both `TypeDeclarationOptions` and `SchemaObjectType` provides `IsAbstract` property, their values should be identical. Maybe we shouldn't even have one of these properties

Reply:
Moved the `IsAbstract` property to the base type TypeProvider and ensured the value is identical with the `TypeDeclarationOptions`